test/contrib/automotive/xcp/xcp_scanner.uts

```diff
@@ -0,0 +1,201 @@
+% Regression tests for the XCP_CAN
+~ python3_only vcan_socket needs_root linux
+
+# More information at http://www.secdev.org/projects/UTscapy/
+
+############
+############
+
++ Basic operations
+= Imports
+
+import threading
+import time
+from subprocess import call
+
+import six
+
+from scapy.consts import LINUX
+from scapy.contrib.automotive.xcp.cto_commands_master import Connect, TransportLayerCmd, TransportLayerCmdGetSlaveId
+from scapy.contrib.automotive.xcp.scanner import XCPOnCANScanner
+from scapy.contrib.automotive.xcp.xcp import CTORequest, XCPOnCAN, CTOResponse, ConnectPositiveResponse, TransportLayerCmdGetSlaveIdResponse, GenericResponse
+from scapy.contrib.cansocket_python_can import CANSocket
+from scapy.main import load_layer
+
+= Load module
+
+load_layer("can")
+load_layer("automotive.xcp")
+
+= Global variables
+
+iface0 = "vcan0"
+
+= Initialize a virtual CAN interface
+
+def set_up_virtual_python_can_socket(iface):
+    return CANSocket(bustype='virtual', channel=iface, timeout=0.01)
+
+print('setting up CAN')
+
+if 0 != call("cansend %s 000#" % iface0, shell=True):
+    # vcan0 is not enabled
+    if 0 != call("sudo modprobe vcan", shell=True):
+        raise Exception("modprobe vcan failed")
+    if 0 != call("sudo ip link add name %s type vcan" % iface0, shell=True):
+        print("add %s failed: Maybe it was already up?" % iface0)
+    if 0 != call("sudo ip link set dev %s up" % iface0, shell=True):
+        raise Exception("could not bring up %s" % iface0)
+
+if 0 != call("cansend %s 000#" % iface0, shell=True):
+    raise Exception("cansend doesn't work")
+
+print("CAN should work now")
+
+new_can_socket0 = lambda: CANSocket(bustype='virtual', channel=iface0, timeout=0.01)
```

Comment:
I think it would be best here, to fix the sockets for this file to CANSocket(**bustype='virtual'**, ...)
This would require only to remove the next two functions but then even more preparation functions can be removed.

This would make the test faster, easier and more reliable. The sockets which communicate with real (virtual) busses are tested somewhere else. So just additional code and runtime overhead if used here again.
@polybassa What is your opinion about this?